src/crawl4ai_mcp.py

```
@@ -69,18 +69,20 @@ def main():
     # Setup signal handlers for graceful shutdown
     def signal_handler(signum, frame):
-        logger.critical(f"🛑 SIGNAL RECEIVED: {signum}")
-        logger.critical(f"🛑 Signal frame: {frame}")
-        logger.critical(f"🛑 Current stack trace:")
+        logger.critical("🛑 SIGNAL RECEIVED: %s", signum)
+        logger.critical("🛑 Signal frame: %s", frame)
+        logger.critical("🛑 Current stack trace:")
         for line in traceback.format_stack():
             logger.critical(line.strip())
-        print(f"\n🛑 Received signal {signum}, shutting down gracefully...")
+        print("\n🛑 Received signal %s, shutting down gracefully..." % signum)
```

Ruff (0.12.2)

Use format specifiers instead of percent format

Replace with format specifiers

(UP031)